examples/kalshi-polymarket-vwap/src/execution_phase.rs

```diff
@@ -0,0 +1,143 @@
+    );
+
+    // Check if the market request was successful
+    if !kalshi_market_response.is_ok() {
+        elog!(
+            "market HTTP Response was rejected: {} - {}",
+            kalshi_market_response.status,
+            String::from_utf8(kalshi_market_response.bytes)?
+        );
+        Process::error("Error while fetching market information".as_bytes());
+        return Ok(());
+    }
+
+    // Parse market informationmarket_response
+    let kalshi_market_data =
+        serde_json::from_slice::<KalshiMarketResponse>(&kalshi_market_response.bytes)?;
+    log!(
+        "Fetched Kalshi Price (YES BID): {} cents with volume {}",
+        kalshi_market_data.market.yes_bid_dollars,
+        kalshi_market_data.market.volume
+    );
+
+    let kalshi_yes_bid_dollars = kalshi_market_data.market.yes_bid_dollars.parse::<f64>()?;
+
+    // Step 2: Fetch PolyMarket market data (yes outcome price and volume)
+    let polymarket_market_response = http_fetch(
+        format!(
+            "https://gamma-api.polymarket.com/markets/{}",
+            market_tickers[1]
+        ),
+        None,
+    );
+
+    // Check if the market request was successful
+    if !polymarket_market_response.is_ok() {
+        elog!(
+            "market HTTP Response was rejected: {} - {}",
+            polymarket_market_response.status,
+            String::from_utf8(polymarket_market_response.bytes)?
+        );
+        Process::error("Error while fetching market information".as_bytes());
+        return Ok(());
+    }
+
+    // Parse market informationmarket_response
+    let poly_market_market_data =
+        serde_json::from_slice::<PolyMarketMarket>(&polymarket_market_response.bytes)?;
+    let outcome_prices_array: Vec<String> =
+        serde_json::from_str(&poly_market_market_data.outcome_prices)?;
+    let polymarket_yes_price = outcome_prices_array[0].parse::<f64>()?; // 0 = yes price
+
+    log!(
+        "Fetched Price (YES PRICE): {} cents with volume {}",
+        polymarket_yes_price,
+        poly_market_market_data.volume
+    );
+
+    let poly_market_volume = poly_market_market_data.volume.parse::<f64>()?;
+
+    // Step 3: Calculate volume-weighted average price between Kalshi and PolyMarket
+    let total_volume = kalshi_market_data.market.volume as f64 + poly_market_volume;
+
+    if total_volume == 0.0 {
+        elog!("Total volume is zero, cannot calculate volume weighted average price");
+        Process::error("Error: Total volume is zero".as_bytes());
+        return Ok(());
+    }
+
+    // Calculate weighted prices: price × volume for each platform
+    let kalshi_weighted_price = kalshi_yes_bid_dollars * (kalshi_market_data.market.volume as f64);
+    let polymarket_weighted_price = polymarket_yes_price * poly_market_volume;
+
+    // Final volume-weighted average: (sum of weighted prices) / (total volume)
+    let volume_weighted_average_price =
+        (kalshi_weighted_price + polymarket_weighted_price) / total_volume;
+
```

Comment:
Depending on the people who want it- it might be best to convert the floats to integers before doing these operations. As float operations DO NOT guarantee precision. And then if they need them in float form we can convert it back.